.github/workflows/test.yml

```diff
@@ -0,0 +1,108 @@
+name: test
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
+  # Test the Android ds2 binary on an emulator.
+  android-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # We don't currently have a good way of running arm64-v8a tests on the
+        # available GitHub test runners.
+        abi: ['x86_64']
+
+    env:
+      ANDROID_API_LEVEL: 28
+      DEBUG_PORT: 5432
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: seanmiddleditch/gha-setup-ninja@master
+
+      # Free up some disk space by removing some things we don't use. Without
+      # this step, we exhaust disk space while installing Android the emulator
+      # image in the setup-ndk step below.
+      - name: free-up disk space
+        run: |
+          df -h /home                                                 && \
+          sudo apt purge -q dotnet-* netstandard-* ghc ghc-* boost-*  && \
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"                         && \
```

Comment:
This seems ... sketchy. This is the tools cache, are we not using any tools from here?

Reply:
I don't love it, but the jobs do all run even after deleting this. It is something like 6GiB. TBH, I haven't tried without it, though, so let me do that and see if just uninstalling the packages frees up sufficient space.

Reply:
I removed this line; will see if the new job run has enough space to pass

Reply:
Yeah, tests failed; not enough disk space to run the emulator without removing the `$AGENT_TOOLSDIRECTORY` path. Let me see if there are some other packages I can uninstall instead to reclaim a bit more space.

Reply:
Latest revision uninstalls a bunch more packages (such as web browsers...) and frees up about 7GiB. Hopefully that's enough for the tests to proceed this time.